queue/group.go

```diff
@@ -0,0 +1,132 @@
+package queue
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/mongodb/amboy"
+	"github.com/pkg/errors"
+)
+
+// NewLocalQueueGroup constructs a new local queue group. If ttl is 0, the collection will not be
+// TTLed except when the client explicitly calls Prune.
+func NewLocalQueueGroup(ctx context.Context, constructor QueueConstructor, ttl time.Duration) (amboy.QueueGroup, error) {
+	if constructor == nil {
+		return nil, errors.New("must pass a constructor")
+	}
+	if ttl < 0 {
+		return nil, errors.New("ttl must be greater than or equal to 0")
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	g := &localQueueGroup{
+		canceler:    cancel,
+		queues:      map[string]amboy.Queue{},
+		constructor: constructor,
+		ttlMap:      map[string]time.Time{},
+		ttl:         ttl,
+	}
+	if ttl > 0 {
+		go func(ctx context.Context) {
+			ticker := time.NewTicker(ttl)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					g.Prune()
+				}
+			}
+		}(ctx)
+	}
+	return g, nil
+}
+
+// QueueConstructor is a function passed by the client which makes a new queue.
+type QueueConstructor func(ctx context.Context) (amboy.Queue, error)
```

Comment:
should we move this file to a different package? the name is weird, as most queue constructors have a different signature

Reply:
Done